### PR TITLE
refactor(node): Introduce `QueueStep`

### LINKED ISSUE #1987: Unify build_journal func in `process_queue`/`calculate_gas_info_impl`
### File Location(s)

_No response_

### Proposal

consider to introduce new common function. After #1959

### PR DESCRIPTION
**Pull request by ark0f**
Part of #1987

### CODE REVIEW

File: pallets/gear/src/journal_builder.rs [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+    common::{ExecutableActorData, JournalNote, PrechargedDispatch},
+    configs::BlockConfig,
+    ContextChargedForInstrumentation,
+};
+use frame_support::traits::{Get, Randomness};
+use gear_core::{ids::ProgramId, message::StoredDispatch};
+use parity_scale_codec::Encode;
+use sp_runtime::traits::UniqueSaturatedInto;
+
+pub(crate) struct JournalBuilder<'a, T: Config, F> {
```
Comment (grishasobol) [inline pallets/gear/src/journal_builder.rs:32]: Completelly disagree wiith naming. This one is just another layer around smart-contract execution. So, name must be something like prepare_and_proccess_message_execution().

File: pallets/gear/src/mock.rs [hunk 2]
```diff
@@ -151,20 +151,43 @@ parameter_types! {
...
     pub const OutgoingLimit: u32 = 1024;
     pub ReserveThreshold: BlockNumber = 1;
-    pub GearSchedule: pallet_gear::Schedule<Test> = <pallet_gear::Schedule<Test>>::default();
     pub RentFreePeriod: BlockNumber = 1_000;
     pub RentCostPerBlock: Balance = 11;
     pub ResumeMinimalPeriod: BlockNumber = 100;
     pub ResumeSessionDuration: BlockNumber = 1_000;
 }
 
+static SCHEDULE: spin::Mutex<Option<Schedule<Test>>> = spin::Mutex::new(None);
```
Comment (grishasobol) [inline pallets/gear/src/mock.rs:160]: Could you please explain this changes?

**Review by grishasobol — CHANGES_REQUESTED** [review]

File: pallets/gear/src/mock.rs (hunk 2 above)
Comment (ark0f) [reply pallets/gear/src/mock.rs:160]: Just check tests

File: pallets/gear/src/journal_builder.rs (hunk 1 above)
Comment (ark0f) [reply pallets/gear/src/journal_builder.rs:32]: Layer that builds journal list as the result

Comment (grishasobol) [reply pallets/gear/src/journal_builder.rs:32]: Is not an argument

File: pallets/gear/src/queue.rs [hunk 3]
```diff
@@ -17,6 +17,149 @@
...
+        let memory_pages = Pallet::<T>::get_and_track_memory_pages(
+            ext_manager,
+            program_id,
+            &context.actor_data().pages_with_data,
+            lazy_pages_enabled,
+        )
+        .ok_or(QueueStepError::NoMemoryPages)?;
+
+        let (random, bn) = T::Randomness::random(dispatch_id.as_ref());
+        let origin = ProgramId::from_origin(external.into_origin());
```
Comment (SabaunT) [inline pallets/gear/src/queue.rs:150]: gr_origin is removed, so do we need this though?

**Review by SabaunT — CHANGES_REQUESTED** [review]
Good job :+1: 
I'd wait for @gshep and @breathx 

Also remove please data query for message origin.

File: pallets/gear/src/queue.rs (hunk 3 above)
Comment (gshep) [reply pallets/gear/src/queue.rs:150]: `gr_origin` is a user space call but on the backend side we could operate with the message origin

Comment (SabaunT) [reply pallets/gear/src/queue.rs:150]: do we really? I see not usage of `origin` field in `ProcessorContext`, except for `gr_origin`

**Review by SabaunT — APPROVED** [review]

File: pallets/gear/src/queue.rs (hunk 3 above)
Comment (breathx) [reply pallets/gear/src/queue.rs:150]: > `gr_origin` is a user space call but on the backend side we may operate with the message origin

Totally unused and shouldn't be used further

**Comment by SabaunT** [comment]
Don't forget to remove `origin` setting!

**Review by NikVolf — APPROVED** [review]

**Comment by NikVolf** [comment]
@grishasobol re-review pls

**Review by grishasobol — APPROVED** [review]

**Pull request merged**